File: packages/shared/ts/enums.ts
export enum UserRole {
  Student = "student",
  Instructor = "instructor",
  Admin = "admin",
}

export enum QuestionType {
  MCQ = "mcq",
  TrueFalse = "true_false",
  Numeric = "numeric",
  ShortAnswer = "short_answer",
  LongForm = "long_form",
}

export enum SessionStatus {
  Created = "created",
  InProgress = "in_progress",
  Submitted = "submitted",
  Graded = "graded",
  Expired = "expired",
}

export enum GradingMethod {
  Deterministic = "deterministic",
  LLM = "llm",
  Manual = "manual",
  Fallback = "fallback",
}

export enum ExamMode {
  Practice = "practice",
  Mock = "mock",
  HighStakes = "high_stakes",
}

export enum DocumentStatus {
  Pending = "pending",
  Processing = "processing",
  Ready = "ready",
  Failed = "failed",
}

File: packages/shared/python/enums.py
from enum import StrEnum


class UserRole(StrEnum):
    STUDENT = "student"
    INSTRUCTOR = "instructor"
    ADMIN = "admin"


class QuestionType(StrEnum):
    MCQ = "mcq"
    TRUE_FALSE = "true_false"
    NUMERIC = "numeric"
    SHORT_ANSWER = "short_answer"
    LONG_FORM = "long_form"


class SessionStatus(StrEnum):
    CREATED = "created"
    IN_PROGRESS = "in_progress"
    SUBMITTED = "submitted"
    GRADED = "graded"
    EXPIRED = "expired"


class GradingMethod(StrEnum):
    DETERMINISTIC = "deterministic"
    LLM = "llm"
    MANUAL = "manual"
    FALLBACK = "fallback"


class ExamMode(StrEnum):
    PRACTICE = "practice"
    MOCK = "mock"
    HIGH_STAKES = "high_stakes"


class DocumentStatus(StrEnum):
    PENDING = "pending"
    PROCESSING = "processing"
    READY = "ready"
    FAILED = "failed"


class GradeStatus(StrEnum):
    PENDING = "pending"
    GRADED = "graded"
    FLAGGED = "flagged"
    MANUAL_REVIEW = "manual_review"
